Parameter 'term' implicitly has an 'any' type

### Expected behaviour
AOT Build should compile without errors.

### Actual behaviour
When using TSLint and the --noImplicitAny  [compiler option](https://www.typescriptlang.org/docs/handbook/compiler-options.html), AOT Build fails with the following error.

`ERROR in node_modules/@ng-select/ng-select/ng-select/ng-select.component.d.ts(34,25): error TS7006: Parameter 'term' implicitly has an 'any' type
`

This is a minor issue that can be fixed by replacing a single line in "ng-select.component.d.ts":

addTag: boolean | ((term) => NgOption); // Current
addTag: boolean | ((term: any) => NgOption); //Suggested Change

Thank you for the great plugin!!

### More Info
ng-select version: 0.14.4

--- a/src/ng-select/items-list.ts
+++ b/src/ng-select/items-list.ts
@@ -43,7 +43,7 @@ export class ItemsList {
         item.selected = true;
     }
 
-    findItem(value, bindValue: string): NgOption {
+    findItem(value: any, bindValue: string): NgOption {
         if (!value) {
             return null;
         }
@@ -109,7 +109,7 @@ export class ItemsList {
         this._markedIndex = this.filteredItems.indexOf(item);
     }
 
-    markSelectedOrDefault(markDefault) {
+    markSelectedOrDefault(markDefault: boolean) {
         if (this.filteredItems.length === 0) {
             return;
         }
@@ -139,7 +139,7 @@ export class ItemsList {
         }
     }
 
-    private getDefaultFilterFunc(term) {
+    private getDefaultFilterFunc(term: string) {
         return (option: NgOption) => {
             return searchHelper.stripSpecialChars(option.label || '')
                 .toUpperCase()

--- a/src/ng-select/ng-select.component.html
+++ b/src/ng-select/ng-select.component.html
@@ -21,7 +21,7 @@
             <input #filterInput
                    type="text"
                    [value]="filterValue"
-                   (input)="onFilter($event)"
+                   (input)="onFilter(filterInput.value)"
                    (focus)="onInputFocus($event)"
                    (blur)="onInputBlur($event)"
                    (change)="$event.stopPropagation()"

--- a/src/ng-select/ng-select.component.spec.ts
+++ b/src/ng-select/ng-select.component.spec.ts
@@ -492,7 +492,7 @@ describe('NgSelectComponent', function () {
 
         describe('tab', () => {
             it('should close dropdown when there are no items', fakeAsync(() => {
-                fixture.componentInstance.select.onFilter({ target: { value: 'random stuff' } });
+                fixture.componentInstance.select.onFilter('random stuff');
                 tick(200);
                 triggerKeyDownEvent(getNgSelectElement(fixture), KeyCode.Space);
                 triggerKeyDownEvent(getNgSelectElement(fixture), KeyCode.Tab);
@@ -755,7 +755,7 @@ describe('NgSelectComponent', function () {
                 </ng-select>`);
 
             tickAndDetectChanges(fixture);
-            fixture.componentInstance.select.onFilter({ target: { value: 'new tag' } });
+            fixture.componentInstance.select.onFilter('new tag');
             tickAndDetectChanges(fixture);
             triggerKeyDownEvent(getNgSelectElement(fixture), KeyCode.Enter);
             expect(fixture.componentInstance.selectedCity.name).toBe('new tag');
@@ -772,7 +772,7 @@ describe('NgSelectComponent', function () {
                 </ng-select>`);
 
             tickAndDetectChanges(fixture);
-            fixture.componentInstance.select.onFilter({ target: { value: 'custom tag' } });
+            fixture.componentInstance.select.onFilter('custom tag');
             tickAndDetectChanges(fixture);
             triggerKeyDownEvent(getNgSelectElement(fixture), KeyCode.Enter);
             expect(<any>fixture.componentInstance.selectedCity).toEqual(jasmine.objectContaining({
@@ -822,7 +822,7 @@ describe('NgSelectComponent', function () {
                 </ng-select>`);
 
             tick(200);
-            fixture.componentInstance.select.onFilter({ target: { value: 'vilnius' } });
+            fixture.componentInstance.select.onFilter('vilnius');
             tick(200);
 
             const result = [jasmine.objectContaining({
@@ -861,7 +861,7 @@ describe('NgSelectComponent', function () {
                     [(ngModel)]="selectedCity">
                 </ng-select>`);
 
-            fixture.componentInstance.select.onFilter({ target: { value: 'vilnius' } });
+            fixture.componentInstance.select.onFilter('vilnius');
             tickAndDetectChanges(fixture);
             expect(fixture.componentInstance.select.filterValue).toBe(null);
         }));
@@ -875,7 +875,7 @@ describe('NgSelectComponent', function () {
                 </ng-select>`);
 
             tick(200);
-            fixture.componentInstance.select.onFilter({ target: { value: 'pab' } });
+            fixture.componentInstance.select.onFilter('pab');
             tick(200);
 
             const result = jasmine.objectContaining({
@@ -896,7 +896,7 @@ describe('NgSelectComponent', function () {
                 </ng-select>`);
 
             tick(200);
-            fixture.componentInstance.select.onFilter({ target: { value: 'pab' } });
+            fixture.componentInstance.select.onFilter('pab');
             tick(200);
             expect(fixture.componentInstance.select.itemsList.markedItem).toEqual(undefined)
         }));
@@ -932,13 +932,13 @@ describe('NgSelectComponent', function () {
                     expect(term).toBe('vilnius');
                 });
                 tick(200);
-                fixture.componentInstance.select.onFilter({ target: { value: 'vilnius' } });
+                fixture.componentInstance.select.onFilter('vilnius');
                 tickAndDetectChanges(fixture);
             }));
 
             it('should mark first item when typeahead results are loaded', fakeAsync(() => {
                 fixture.componentInstance.customFilter.subscribe();
-                fixture.componentInstance.select.onFilter({ target: { value: 'buk' } });
+                fixture.componentInstance.select.onFilter('buk');
                 fixture.componentInstance.cities = [{ id: 4, name: 'Bukiskes' }];
                 tickAndDetectChanges(fixture);
                 triggerKeyDownEvent(getNgSelectElement(fixture), KeyCode.Enter);
@@ -950,7 +950,7 @@ describe('NgSelectComponent', function () {
             it('should not mark first item when typeahead results are loaded', fakeAsync(() => {
                 fixture.componentInstance.select.markFirst = false;
                 fixture.componentInstance.customFilter.subscribe();
-                fixture.componentInstance.select.onFilter({ target: { value: 'buk' } });
+                fixture.componentInstance.select.onFilter('buk');
                 fixture.componentInstance.cities = [{ id: 4, name: 'Bukiskes' }];
                 tickAndDetectChanges(fixture);
                 triggerKeyDownEvent(getNgSelectElement(fixture), KeyCode.Enter);
@@ -959,7 +959,7 @@ describe('NgSelectComponent', function () {
 
             it('should start and stop loading indicator', fakeAsync(() => {
                 fixture.componentInstance.customFilter.subscribe();
-                fixture.componentInstance.select.onFilter({ target: { value: 'buk' } });
+                fixture.componentInstance.select.onFilter('buk');
                 expect(fixture.componentInstance.select.isLoading).toBeTruthy();
                 fixture.componentInstance.cities = [{ id: 4, name: 'Bukiskes' }];
                 tickAndDetectChanges(fixture);

--- a/src/ng-select/ng-select.component.ts
+++ b/src/ng-select/ng-select.component.ts
@@ -58,7 +58,7 @@ export class NgSelectComponent implements OnInit, OnDestroy, OnChanges, AfterVie
     @ViewChild(VirtualScrollComponent) dropdownList: VirtualScrollComponent;
     @ViewChild('dropdownPanel') dropdownPanel: ElementRef;
     @ContentChildren(NgOptionComponent, { descendants: true }) ngOptions: QueryList<NgOptionComponent>;
-    @ViewChild('filterInput') filterInput;
+    @ViewChild('filterInput') filterInput: ElementRef;
 
     // inputs
     @Input() items = [];
@@ -68,13 +68,13 @@ export class NgSelectComponent implements OnInit, OnDestroy, OnChanges, AfterVie
     @Input() markFirst = true;
     @Input() disableVirtualScroll = false;
     @Input() placeholder: string;
-    @Input() notFoundText;
-    @Input() typeToSearchText;
-    @Input() addTagText;
-    @Input() loadingText;
-    @Input() clearAllText;
+    @Input() notFoundText: string;
+    @Input() typeToSearchText: string;
+    @Input() addTagText: string;
+    @Input() loadingText: string;
+    @Input() clearAllText: string;
     @Input() dropdownPosition: 'bottom' | 'top' = 'bottom';
-    @Input() appendTo;
+    @Input() appendTo: string;
 
     @Input()
     @HostBinding('class.typeahead') typeahead: Subject<string>;
@@ -83,7 +83,7 @@ export class NgSelectComponent implements OnInit, OnDestroy, OnChanges, AfterVie
     @HostBinding('class.ng-multiple') multiple = false;
 
     @Input()
-    @HostBinding('class.taggable') addTag: boolean | ((term) => NgOption) = false;
+    @HostBinding('class.taggable') addTag: boolean | ((term: string) => NgOption) = false;
 
     @Input()
     @HostBinding('class.searchable') searchable = true;
@@ -124,7 +124,7 @@ export class NgSelectComponent implements OnInit, OnDestroy, OnChanges, AfterVie
     private disposeDocumentClickListener = () => { };
     private disposeDocumentResizeListener = () => { };
 
-    clearItem = (item) => {
+    clearItem = (item: any) => {
         const option = this.itemsList.items.find(x => x.value === item);
         this.unselect(option);
     };
@@ -360,7 +360,7 @@ export class NgSelectComponent implements OnInit, OnDestroy, OnChanges, AfterVie
         return empty && this.isTypeahead && !this.filterValue && !this.isLoading;
     }
 
-    onFilter($event) {
+    onFilter(term: string) {
         if (!this.searchable) {
             return;
         }
@@ -369,7 +369,7 @@ export class NgSelectComponent implements OnInit, OnDestroy, OnChanges, AfterVie
             this.open();
         }
 
-        this.filterValue = $event.target.value;
+        this.filterValue = term;
 
         if (this.isTypeahead) {
             this.isLoading = true;
@@ -419,7 +419,7 @@ export class NgSelectComponent implements OnInit, OnDestroy, OnChanges, AfterVie
     private setItemsFromNgOptions() {
         this.bindLabel = this.bindLabel || this._defaultLabel;
         this.bindValue = this.bindValue || this._defaultValue;
-        const handleNgOptions = (options) => {
+        const handleNgOptions = (options: QueryList<NgOptionComponent>) => {
             this.items = options.map(option => ({
                 value: option.value,
                 label: option.elementRef.nativeElement.innerHTML
@@ -438,7 +438,7 @@ export class NgSelectComponent implements OnInit, OnDestroy, OnChanges, AfterVie
     }
 
     private handleDocumentClick() {
-        const handler = ($event) => {
+        const handler = ($event: any) => {
             // prevent close if clicked on select
             if (this.elementRef.nativeElement.contains($event.target)) {
                 return;
@@ -479,7 +479,7 @@ export class NgSelectComponent implements OnInit, OnDestroy, OnChanges, AfterVie
         if (this.appendTo === 'body') {
             document.body.appendChild(this.dropdownPanel.nativeElement);
         } else {
-            const parent: HTMLElement = document.querySelector(this.appendTo);
+            const parent = document.querySelector(this.appendTo);
             if (!parent) {
                 throw new Error(`appendTo selector ${this.appendTo} did not found any parent element`)
             }
@@ -508,7 +508,7 @@ export class NgSelectComponent implements OnInit, OnDestroy, OnChanges, AfterVie
             return;
         }
 
-        const validateBinding = (item) => {
+        const validateBinding = (item: any) => {
             if (item instanceof Object && this.bindValue) {
                 throw new Error('Binding object with bindValue is not allowed.');
             }

--- a/src/ng-select/ng-select.component.ts
+++ b/src/ng-select/ng-select.component.ts
@@ -61,7 +61,7 @@ export class NgSelectComponent implements OnInit, OnDestroy, OnChanges, AfterVie
     @ViewChild('filterInput') filterInput: ElementRef;
 
     // inputs
-    @Input() items = [];
+    @Input() items: any[] = [];
     @Input() bindLabel: string;
     @Input() bindValue: string;
     @Input() clearable = true;
@@ -114,7 +114,7 @@ export class NgSelectComponent implements OnInit, OnDestroy, OnChanges, AfterVie
     isLoading = false;
     filterValue: string = null;
 
-    private _ngModel = null;
+    private _ngModel: any = null;
     private _simple = false;
     private _defaultLabel = 'label';
     private _defaultValue = 'value';

